src/Main.hs

```diff
@@ -101,6 +103,25 @@ main = withUtf8 $ withCP65001 $ do
         corpus <- readIORef env.corpusRef
         saveTxs env (dir </> "coverage") (snd <$> Set.toList corpus)
 
+      let isLargeOrSolved Solved = True
+          isLargeOrSolved (Large _) = True
+          isLargeOrSolved _ = False
+      measureIO cfg.solConf.quiet "Saving foundry reproducers" $ do
+        let foundryDir = dir </> "foundry"
+        liftIO $ createDirectoryIfMissing True foundryDir
+        forM_ tests $ \test ->
+          case (test.testType, test.state) of
+            (AssertionTest{}, state) | isLargeOrSolved state ->
+              do
+              let
+                reproducerHash = (show . abs . hash) test.reproducer
+                fileName = foundryDir </> "Test." ++ reproducerHash <.> "sol"
+                content = foundryTest cliSelectedContract test
+              liftIO $ writeFile fileName (TL.unpack content)
+            _ -> pure ()
+
+      -- TODO: We use the corpus dir to save coverage reports which is confusing.
```

Comment:
what does this floating comment mean? Is it from a mismerge?

Reply:
Yes, my fault, let me remove it.

Reply:
Done!